tests/rados/test_mon_mgr_weight.py

```diff
@@ -0,0 +1,148 @@
+            time.sleep(10)
+
+            # check if the value was set correctly
+            out = rados_obj.run_ceph_command(cmd="ceph mon dump", client_exec=True)
+            mon_list = out["mons"]
+
+            for mon in mon_list:
+                if int(mon["rank"]) == 0:
+                    break
+
+            log.info(f"Monitor details \n: {mon}")
+            if not mon["weight"] == weight:
+                log.error(f"Mon weight for active Mon {active_mon} not set properly")
+                raise Exception(
+                    f"Mon weight for active Mon {active_mon} not set properly"
+                )
+
+            log.info(
+                f"Mon weight for Active mon {active_mon} set to {weight} successfully"
+            )
+
+            # restart Mgr services and monitor cluster for 120 secs for any crash
+            if not rados_obj.restart_daemon_services(daemon="mgr"):
+                log.error("Manager orch service did not restart as expected")
+                raise Exception("Manager orch service did not restart as expected")
+
+            mgr_status, mgr_status_desc = rados_obj.get_daemon_status(
+                daemon_type="mgr", daemon_id=active_mgr
+            )
+            if mgr_status != 1 or mgr_status_desc != "running":
+                log.error(
+                    "Manager for whom mon-weight was modified is not in running state post "
+                    "restart"
+                )
+                raise Exception("Manager with modified weight is not in running state")
+
+            timeout_time = datetime.datetime.now() + datetime.timedelta(seconds=120)
+            while datetime.datetime.now() < timeout_time:
+                # list any reported crashes on the cluster, should not have any
+                crash_out = rados_obj.run_ceph_command(
+                    cmd="ceph crash ls-new", client_exec=True
+                )
+                if len(crash_out) != 0:
+                    log.error(
+                        f"Verification failed, crash observed on the cluster : \n {crash_out}"
+                    )
+                    raise Exception(
+                        "Verification failed, crash observed on the cluster"
+                    )
+                log.info(
+                    "No crashes reported on the cluster, proceeding to check health status"
+                )
+
+                # fetch ceph health detail
+                health_detail, _ = cephadm.shell(args=["ceph health detail"])
+                assert "no active mgr" not in health_detail
+                time.sleep(30)
+
+    except Exception as e:
+        log.error(f"Failed with exception: {e.__doc__}")
+        log.exception(e)
+        return 1
+    finally:
+        log.info(
+            "\n \n ************** Execution of finally block begins here *************** \n \n"
+        )
+        # reset monitor weight to 0
+        set_weight_cmd = f"ceph mon set-weight {active_mon} 0"
```

Comment:
please check if the var `active_mon` is set before trying to use it here.

Reply:
Done, please review